game-core/src/test/java/games/strategy/triplea/delegate/DiceRollTest.java

```diff
@@ -741,4 +750,61 @@ void testSbrRolls() {
     assertThat(StrategicBombingRaidBattle.getSbrRolls(bomber, british), is(2));
     assertThat(StrategicBombingRaidBattle.getSbrRolls(bomber, british), is(2));
   }
+
+  @Nested
+  @ExtendWith(MockitoExtension.class)
+  class SortAaHighToLowTest {
+
+    @Mock private Unit unit1;
+    @Mock private Unit unit2;
+    @Mock private Unit unit3;
+    @Mock private Unit unit4;
+    @Mock private Unit unit5;
+
+    private final Map<Unit, Tuple<Integer, Integer>> unitPowerAndRollsMap = new HashMap<>();
+    private final List<Unit> units = new ArrayList<>();
+
+    @BeforeEach
+    void setUp() {
+      units.addAll(List.of(unit1, unit2, unit3, unit4, unit5));
+      for (final var unit : units) {
+        final UnitType unitTypeMock = mock(UnitType.class);
+        final UnitAttachment unitAttachment = mock(UnitAttachment.class);
+        when(unitTypeMock.getAttachment(any())).thenReturn(unitAttachment);
+        when(unit.getType()).thenReturn(unitTypeMock);
+      }
+    }
+
+    @Test
+    void testAttacking() {
+      int index = 4;
+      for (final var unit : units) {
+        when(((UnitAttachment) unit.getType().getAttachment("")).getOffensiveAttackAa(any()))
+            .thenReturn(index / 2);
```

Comment:
What does 'index / 2' represent? Would that question be answered by extracting the value to a variable?

Reply:
Not really, in order to test the `getFirst() == 0` cases, I had to introduce duplicate keys somehow to get this edge case covered.

A comment would help though IMO